pkgs/top-level/lua-packages.nix

```diff
@@ -229,6 +229,12 @@ let
     };
     buildInputs = [ unzip ];
 
+    preBuild = ''
+      makeFlagsArray=(CC=$CC);
+    '';
+
+    buildFlags = if stdenv.isDarwin then "macosx" else "";
```

Comment:
I dont know if all lua packages have a macosx target and set CC=gcc. I am very new to nix and actually try to get neovim running. Maybe some lua maintainer can help? I can then try to implement the correct solution.